dao/src/main/java/com/fitgoal/dao/mybatis/mappers/UserMapper.java

```diff
@@ -0,0 +1,25 @@
+package com.fitgoal.dao.mybatis.mappers;
+
+import com.fitgoal.dao.domain.UserDto;
+import org.apache.ibatis.annotations.Insert;
+import org.apache.ibatis.annotations.Options;
+import org.apache.ibatis.annotations.Select;
+import org.apache.ibatis.annotations.Update;
+
+import java.util.Optional;
+
+public interface UserMapper {
+
+    @Insert("Insert into users(email, password, link, active) values (#{email}, #{password}, #{link}, #{active})")
+    @Options(useGeneratedKeys = true, keyProperty = "id")
+    void save(UserDto userDto);
+
+    @Select("Select id, email, password, link, active from users where email = #{userEmail}")
```

Comment:
just wondering about sql injections, does mybatis escapes the injections under the hood?

Reply:
I know that we protect ourselves using the Hibernate validation, but still it's interesting.